src/Sylius/Bundle/SettingsBundle/Model/SettingsInterface.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\SettingsBundle\Model;
+
+use Sylius\Bundle\SettingsBundle\Exception\ParameterNotFoundException;
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+/**
+ * @author Steffen Brem <steffenbrem@gmail.com>
+ */
+interface SettingsInterface extends ResourceInterface, \ArrayAccess, \Countable
+{
+    /**
+     * @return string
+     */
+    public function getSchemaAlias();
+
+    /**
+     * @param string $schemaAlias
+     */
+    public function setSchemaAlias($schemaAlias);
+
+    /**
+     * @return string
+     */
+    public function getNamespace();
+
+    /**
+     * @param string $namespace
+     */
+    public function setNamespace($namespace);
```

Comment:
Hmm yeah I think we could safely remove this from the interface, same for `setSchemaAlias`? Since they should be immutable. Should we keep the setters on the concrete classes? Or should I look into constructor injection and fully remove those setters and use a factory method for this.